docs/elasticsearch-implementation.md

```diff
@@ -0,0 +1,71 @@
+# ElasticSearch
```

Comment:
Can we use the canonical capitalization "Elasticsearch" throughout? [0](https://www.elastic.co/elasticsearch/), [1](https://en.wikipedia.org/wiki/Elasticsearch), [2](https://github.com/elastic/elasticsearch#elasticsearch)